src/main/java/org/reactome/release/resourcechecker/FileResourceChecker.java

```diff
@@ -0,0 +1,148 @@
+		return fileSizeReportJson;
+	}
+
+	default String getFileSizeFound() {
+		long fileSizeInBytes = getFileSize();
+
+		return String.format(
+			"%s (%d bytes)",
+			ByteUnit.getHumanReadableFileSize(fileSizeInBytes),
+			fileSizeInBytes
+		);
+	}
+
+	@Override
+	default boolean resourcePassesAllChecks() {
+		return resourceExists() && isFileSizeAcceptable(getResource().getExpectedFileSizeInBytes());
+	}
+
+	enum ByteUnit {
+		BYTE(0, ""),
+		KILOBYTE(1,"KB"),
+		MEGABYTE(2,"MB"),
+		GIGABYTE(3,"GB");
+
+		private static final double BYTE_UNIT_CONVERSION_FACTOR = 1024.0;
+		private int byteUnitMagnitude;
+		private String byteUnitSuffix;
+
+		ByteUnit(int byteUnitMagnitude, String byteUnitSuffix) {
+			this.byteUnitMagnitude = byteUnitMagnitude;
+			this.byteUnitSuffix = byteUnitSuffix;
+		}
+
+		public int getByteUnitMagnitude() {
+			return this.byteUnitMagnitude;
+		}
+
+		public String getByteUnitSuffix() {
+			return this.byteUnitSuffix;
+		}
+
+		/**
+		 * Converts the file size passed in bytes to the requested ByteUnit.  If the ByteUnit passed is BYTE, there
+		 * will be no change other than receiving the result as a String.
+		 *
+		 * @param fileSizeInBytes Size of the file in bytes
+		 * @param byteUnit Type of unit to retrieve file size as (i.e. BYTE, KILOBYTE, MEGABYTE, GIGABYTE)
+		 * @return Returns the file's size with the appropriate byte unit as a suffix (e.g. calling this method as
+		 * follows - ByteUnit.getFileSizeAs(1536, ByteUnit.MEGABYTE) - will produce the result "1.5 MB")
+		 * @throws IllegalArgumentException Thrown if the value of fileSizeInBytes is less than 0.
+		 */
+		public static String getFileSizeAs(long fileSizeInBytes, ByteUnit byteUnit) {
+			if (fileSizeInBytes < 0) {
+				throw new IllegalArgumentException("The fileSizeInBytes argument must have a positive value");
+			}
+
+			double convertedFileSize = fileSizeInBytes /
+				Math.pow(BYTE_UNIT_CONVERSION_FACTOR, byteUnit.getByteUnitMagnitude());
+			String convertedFileSizeSuffix = byteUnit.getByteUnitSuffix();
+
+			return appendByteUnitSuffixToFileSize(convertedFileSize, convertedFileSizeSuffix);
+		}
+
+		public static String getHumanReadableFileSize(long fileSizeInBytes) {
+			int magnitudeOfByteConversionFactor = 0;
+			double fileSize = fileSizeInBytes;
+			while (fileSize > BYTE_UNIT_CONVERSION_FACTOR) {
+				magnitudeOfByteConversionFactor += 1;
+				fileSize = fileSize / BYTE_UNIT_CONVERSION_FACTOR;
+			}
+
+			return appendByteUnitSuffixToFileSize(
+				fileSize, getSuffixByByteUnitMagnitude(magnitudeOfByteConversionFactor)
+			);
+		}
+
+		private static String getSuffixByByteUnitMagnitude(int byteUnitMagnitude) {
+			return Arrays.stream(ByteUnit.values())
+				.filter(byteUnit -> byteUnit.getByteUnitMagnitude() == byteUnitMagnitude)
+				.findFirst()
+				.orElseThrow(() -> new IllegalArgumentException(
+					"Could not find suffix for the ByteUnit magnitude of " + byteUnitMagnitude
+				))
```

Comment:
Considering that `ByteUnit.values()` is static for the lifespan of the program's execution, is this perhaps a bit excessive? Streaming and filtering? Wouldn't a switch/case be much simpler here, in terms of comprehension and maintenance?

Reply:
Fair point.  A `switch/case` would require hardcoding the individual cases, but I could replace the code with a for loop instead of a stream:

```
for (ByteUnit byteUnit : ByteUnit.values()) {
	if (byteUnit.getByteUnitMagnitude() == byteUnitMagnitude) {
		return byteUnit.getByteUnitSuffix();
	}
}

throw new IllegalArgumentException(
	"Could not find suffix for the ByteUnit magnitude of " + byteUnitMagnitude
);
```

Reply:
I don't think _any_ loop should be necessary here - unless you plan to add/remove ByteUnits on a regular basis.

Reply:
I wanted to keep the information of the magnitude and the suffix for the different units together, but with a `switch/case`, I'd have to hard code the values which would repeat them and do something like this:

```
switch (byteUnitMagnitude) {
	case 0:
		return ByteUnit.BYTE.getByteUnitSuffix();
	case 1:
		return ByteUnit.KILOBYTE.getByteUnitSuffix();
        //...
}
```

I'd prefer to keep the loop so there's no need to do that.  I think the logic is simplified enough with the for-loop.

Reply:
Hmmm are you sure expressions such as `case BYTE:...` won't work? I'm pretty sure you can use `case` directly against the values of an enum - I don't think you need to reference the numeric values.

Reply:
Or you could create a `Map<ByteUnit, String>` and return the string mapped by the given unit. Then you wouldn't need a loop _OR_ or switch/case.

Reply:
That's possible, but the parameter value for the method is an `int`.  There isn't a `ByteUnit` value that can be switched on or mapped.  It would require duplicating information that's already in the `enum`.  The relationship between magnitude and suffix is coded into the constructor of the `enum`.  Using a map, it would be necessary to move that information out of the context of the ByteUnit it's a part of so it would become a `Map<Integer, String>` without the context of the ByteUnit objects it's describing.

Reply:
I now realise that passing in an `int` (which also seems weird to me) is a result of the way `getHumanReadableFileSize` works... I can't help but feel there must be a simpler, non-looping way of doing this... May try to write one later if I get a chance.

Reply:
I found a way to do it without a loop:

```
private static String getSuffixByByteUnitMagnitude(int byteUnitMagnitude) {
	try {
		return ByteUnit.values()[byteUnitMagnitude].getByteUnitSuffix();
	} catch (ArrayIndexOutOfBoundsException e) {
		throw new IllegalArgumentException(
			"Could not find suffix for the ByteUnit magnitude of " + byteUnitMagnitude
		);
	}
}
```